examples/classic/distributed/TeachingConcurrency/teachingConcurrency.qnt

```diff
@@ -0,0 +1,101 @@
+// -*- mode: Bluespec; -*-
+/**
+ * A simple algorithm from Lamport's "Teaching Concurrency" paper:
+ * https://lamport.azurewebsites.net/pubs/teaching-concurrency.pdf
+ *
+ * Run in verifier with:
+ * quint verify --temporal correct teachingConcurrency.qnt
+ */
+
+// An example instance of the algorithm with N=5.
+module teachingConcurrency {
+    import teachingConcurrencyN(NUM_PROCS=5).*
+}
+
+// A concurrency algorithm providing a basic form of mutual
+// exclusion. For details and proofs see section 7.2 of
+// https://lamport.azurewebsites.net/tla/proving-safety.pdf
+module teachingConcurrencyN {
+    // N processes (numbered from 0 to N-1).
+    const NUM_PROCS: int
+    assume nonEmpty = NUM_PROCS >= 1
+
+    var x: List[int]
+    var y: List[int]
+
+    // Set of processes yet to be executed.
+    var pendingProcs: Set[int]
+
+    // Initial state:
+    //  - x and y zeroed for all processes
+    //  - all processes pending scheduling
+    action init = all {
+        x' = zeroes(NUM_PROCS),
+        y' = zeroes(NUM_PROCS),
+        pendingProcs' = 0.to(NUM_PROCS-1),
+    }
+
+    // Stepping action: randomly pick a pending process (if
+    // any) and execute it. Otherwise stutter forever, as
+    // every process has stopped.
+    action step = any {
+        all {
+            not(termination),
+            processUpdate,
+        },
+        all {
+            termination,
+            allStopped,
+        }
+    }
```

Comment:
This specific pattern is exactly an if-then-else. I think it reads much easier as:
```suggestion
    action step = {
        if (termination) {
            allStopped
        } else {    
            processUpdate
        }
    }
```

Reply:
Ah thanks, I didn't realize that; I've directly committed your suggestion.

Speaking of idiomatic patterns, is there a better away to express the termination state, without an explicit stuttering/nop step? I tried avoiding the `allStopped` action, but Apalache (correctly) complains about a deadlock. I didn't see a way to give a hint to the verifier that the execution can be halted at that point (so I don't really care about reassigning all state vars).